storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -335,16 +333,14 @@
         drainer (TransferDrainer.)
         node+port->socket (:cached-node+port->socket worker)
         task->node+port (:cached-task->node+port worker)
-        endpoint-socket-lock (:endpoint-socket-lock worker)
         ]
     (disruptor/clojure-handler
       (fn [packets _ batch-end?]
         (.add drainer packets)
         
         (when batch-end?
-          (read-locked endpoint-socket-lock
-            (let [node+port->socket @node+port->socket]
-              (.send drainer node+port->socket)))
+          (let [node+port->socket @node+port->socket]
+            (.send drainer node+port->socket))
```

Comment:
@nathanmarz, I could be missing something, but I do not think we need a read lock here to protect against the send:

There is no corresponding write-lock around `:cached-node+port->socket`.  (I do not think we need one anyway, since it is updated via a `swap!` in `mk-refresh-connections`, and that should be atomic.)